test/sparse/sparse.jl

```diff
@@ -1843,3 +1843,14 @@ end
     B = A[5:-1:1, 5:-1:1]
     @test issymmetric(B)
 end
+
+@testset "optimizing sparse covariance" begin
+    n = 1000
+    p = 5
+    srand(1)
+    x_sparse = sprand(n, p, .25)
+    x_dense = convert(Matrix{Float64}, x_sparse)
+    cov1 = cov(x_dense)
+    cov2 = cov(x_sparse)
+    @test cov1 ≈ cov2
```

Comment:
It would also be good to test a matrix with `NaN` and another one with `Inf`, to ensure that the result is consistent with the standard method.